src/map.rs

```diff
@@ -426,12 +470,17 @@ pub fn process_loaded_tile_maps(
                     // println!("in object_group {}, object {:?}, grp: {}", object_group.name, &object.tileset_gid, object.gid);
                     let atlas_handle = object
                         .tileset_gid
-                        .and_then(|tileset_gid| texture_atlas_map.get(&tileset_gid));
+                        .and_then(|tileset_gid| texture_atlas_map.get(&(tileset_gid + object.sprite_index.unwrap() - 1))); 
+
+                    let atlas_sprite_index = &map.map.tilesets.iter()
+                        .find(|tileset| Some(tileset.first_gid) == object.tileset_gid)
+                        .and_then(|tileset| if tileset.images.len() > 0 { object.sprite_index } else { Some(0) } );
 
                     let entity = object
                         .spawn(
                             &mut commands,
                             atlas_handle,
+                            *atlas_sprite_index,
```

Comment:
Since some tiles now refer to single-image `TextureAtlas`es, we can no longer assume that TMX sprite index is equivalent to the index in the texture atlas. So, I've added another argument to `spawn` to pass in the atlas index. This may also make it easier to do other kinds of optimizations later? (E.g. splitting or merging tilemaps?)